analysis/study_definition.py

```diff
@@ -1,168 +1,376 @@
+        codelists.eth2001,
+        returning="date",
+        date_format="YYYY",
+        find_last_match_in_period=True,
+    ),
+
+    # Ethnicity no record
+    eth_norecord=patients.with_these_clinical_events(
+        codelists.eth_norecord,
+        returning="date",
+        date_format="YYYY",
+        find_last_match_in_period=True,
+    ),
+
+    # Ethnicity not given - patient refused
+    eth_notgiptref=patients.with_these_clinical_events(
+        codelists.eth_notgiptref,
+        returning="date",
+        date_format="YYYY",
+        find_last_match_in_period=True,
+    ),
+
+    # Ethnicity not stated
+    eth_notstated=patients.with_these_clinical_events(
+        codelists.eth_notstated,
+        returning="date",
+        date_format="YYYY",
+        find_last_match_in_period=True,
+    ),
+
+    # to represent household contact of shielding individual
+    hhld_imdef=patients.with_these_clinical_events(
+        codelists.hhld_imdef,
+        returning="date",
+        date_format="YYYY",
+        find_last_match_in_period=True,
+    ),
+
+    # Immunosuppression diagnosis codes
+    immdx_cov=patients.with_these_clinical_events(
+        codelists.immdx_cov,
+        returning="date",
+        date_format="YYYY",
+        find_last_match_in_period=True,
+    ),
+
+    # Immunosuppression medication codes
+    immrx=patients.with_these_medications(
+        codelists.immrx,
+        returning="date",
+        date_format="YYYY",
+        find_last_match_in_period=True,
+    ),
+
+    # Wider Learning Disability
+    learndis=patients.with_these_clinical_events(
+        codelists.learndis,
+        returning="date",
+        date_format="YYYY",
+        find_last_match_in_period=True,
+    ),
+
+    # Patients in long-stay nursing and residential care
+    longres=patients.with_these_clinical_events(
+        codelists.longres,
+        returning="date",
+        date_format="YYYY",
+        find_last_match_in_period=True,
+    ),
+
+    # Any other ethnicity code
+    non_eth2001=patients.with_these_clinical_events(
+        codelists.non_eth2001,
+        returning="date",
+        date_format="YYYY",
+        find_last_match_in_period=True,
+    ),
+
+    # Lower Risk from COVID-19 codes
+    nonshield=patients.with_these_clinical_events(
+        codelists.nonshield,
+        returning="date",
+        date_format="YYYY",
+        find_last_match_in_period=True,
+    ),
+
+    # No longer a carer codes
+    notcarer=patients.with_these_clinical_events(
+        codelists.notcarer,
+        returning="date",
+        date_format="YYYY",
+        find_last_match_in_period=True,
+    ),
+
+    # Employed by nursing home codes
+    nursehome=patients.with_these_clinical_events(
+        codelists.nursehome,
+        returning="date",
+        date_format="YYYY",
+        find_last_match_in_period=True,
+    ),
+
+    # Pregnancy codes recorded in the 8.5 months before the audit run date
```

Comment:
Maybe we should return YYYY-MM here so we can limit to a shorter period?

Reply:
Shouldn't think it matters too much for checking prevalence?

Reply:
I guess the problem would be if there'd been a change in coding practice in one system over the last few years (e.g. one system used to record pregnancy in a completely different way e.g. in a different table a few years ago) then we might find a difference here that's not relevant to the full definition that only looks for recent codes. But is unlikely! And in the analysis script I've limited this to 2020/21, which is probably a reasonable compromise.

Reply:
Good point.  I reckon limiting to 2020/21 in the notebook is fine for now though.